patches/minecraft/net/minecraft/client/renderer/GameRenderer.java.patch

```diff
@@ -9,6 +9,46 @@
        }
  
     }
+@@ -230,10 +_,10 @@
+             int i = 3;
+             double d1 = d0;
+             if (this.field_78531_r.field_71442_b.func_78749_i()) {
+-               d1 = 6.0D;
++               d1 = this.field_78531_r.field_71439_g.func_110148_a(net.minecraftforge.common.ForgeMod.REACH_DISTANCE.get()).func_111126_e() + 1.0D;
+                d0 = d1;
+             } else {
+-               if (d0 > 3.0D) {
++               if (d0 > this.field_78531_r.field_71439_g.func_110148_a(net.minecraftforge.common.ForgeMod.ATTACK_REACH.get()).func_111126_e()) {
+                   flag = true;
+                }
+ 
+@@ -246,19 +_,21 @@
+             }
+ 
+             Vector3d vector3d1 = entity.func_70676_i(1.0F);
+-            Vector3d vector3d2 = vector3d.func_72441_c(vector3d1.field_72450_a * d0, vector3d1.field_72448_b * d0, vector3d1.field_72449_c * d0);
++            double reach = java.lang.Math.max(d0, this.field_78531_r.field_71439_g.func_110148_a(net.minecraftforge.common.ForgeMod.ATTACK_REACH.get()).func_111126_e());
++            Vector3d vector3d2 = vector3d.func_72441_c(vector3d1.field_72450_a * reach, vector3d1.field_72448_b * reach, vector3d1.field_72449_c * reach);
+             float f = 1.0F;
+-            AxisAlignedBB axisalignedbb = entity.func_174813_aQ().func_216361_a(vector3d1.func_186678_a(d0)).func_72314_b(1.0D, 1.0D, 1.0D);
++            double reachSq = java.lang.Math.max(d1, this.field_78531_r.field_71439_g.func_110148_a(net.minecraftforge.common.ForgeMod.ATTACK_REACH.get()).func_111126_e() * this.field_78531_r.field_71439_g.func_110148_a(net.minecraftforge.common.ForgeMod.ATTACK_REACH.get()).func_111126_e());
```

Comment:
You can just use `reach * reach` instead of grabbing that attribute field again.

Reply:
reach uses d0, which is a different value from reachSq, which uses d1

Reply:
Yeah, but they're set to the same values aren't they?

Reply:
d0 is either the pick reach (5.0) or the creative reach (6.0). d1 is equal to d0 until it gets squared, and then potentially changed to the distance sq from the camera to the block hit result if it's not null. 

Do you think that the distance sq from the camera to the block hit result is going to be the same as d0 squared?

Reply:
I'm using an incorrect given for my claim since d0 shouldn't be a part of the attack reach check. Apologies.